llvm/lib/CodeGen/AsmPrinter/DwarfUnit.h

```diff
@@ -301,6 +301,9 @@ class DwarfUnit : public DIEUnit {
   /// Get context owner's DIE.
   DIE *createTypeDIE(const DICompositeType *Ty);
 
+  /// Returns a unique ID.
+  virtual unsigned getUniqueID() const = 0;
+
```

Comment:
Given that these IDs would be in distinct collections for type units and compile units, is it useful for this to be a virtual function? What code is generic over CU and TU and uses the unique ID effectively?

Reply:
It's used in DWARF5AccelTableData constructor this way we don't care if it's for CU or TU. Just as long as it's unique ID of whatever unit is passed in. It's also used in addTypeUnitSymbol when recording Label and TU to which it belongs. Although there it's TU specific.

Reply:
Changed to passing UniqueID